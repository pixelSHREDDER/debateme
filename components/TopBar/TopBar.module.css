.title {
  color: light-dark(var(--mantine-color-black), var(--mantine-color-white));
  font-size: rem(100px);
  font-weight: 900;
  letter-spacing: rem(-2px);

  @media (max-width: $mantine-breakpoint-md) {
    font-size: rem(50px);
  }
}

.header {
  height: rem(56px);
  margin-bottom: rem(120px);
  /* background-color: var(--mantine-color-body); */
  background-color: light-dark(var(--mantine-primary-color-3), var(--mantine-color-dark-4));
  border-bottom: rem(1px) solid light-dark(var(--mantine-color-gray-3), var(--mantine-color-dark-4));
}

.inner {
  height: rem(56px);
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.link {
  text-transform: uppercase;
  font-size: var(--mantine-font-size-xs);
  color: light-dark(var(--mantine-color-gray-6), var(--mantine-color-dark-1));
  padding: rem(7px) var(--mantine-spacing-sm);
  font-weight: 700;
  border-bottom: rem(2px) solid transparent;
  transition:
    border-color 100ms ease,
    color 100ms ease;

  @mixin hover {
    color: light-dark(var(--mantine-color-black), var(--mantine-color-white));
    text-decoration: none;
  }

  &[data-active] {
    color: light-dark(var(--mantine-color-black), var(--mantine-color-white));
    border-bottom-color: var(--mantine-primary-color-6);
  }
}