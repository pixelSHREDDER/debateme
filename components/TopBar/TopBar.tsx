'use client'

import { useUser } from "@auth0/nextjs-auth0/client"
import Link from "next/link"

export default function TopBar() {
  const { error, isLoading, user } = useUser()

const userChunk = () => {
  if (isLoading) return 'loading user....'
  if (error) return 'error'
  if (user) return (
    <>
      <a href="/user/profile">Hi, {user.name}!</a>&nbsp;|&nbsp;<a href="/api/auth/logout">Logout</a>
    </>
  )
  return <a href="/api/auth/login">Login</a>
}

  return (
    <nav className="w-full h-8 sticky bg-black text-white mb-1 p-1 z-50">
      <Link href="/">Home</Link>&nbsp;|&nbsp;
      <Link href="/debate">Debates</Link>&nbsp;|&nbsp;
      {userChunk()}
    </nav>
  )
}