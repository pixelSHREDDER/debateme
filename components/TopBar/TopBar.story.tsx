import TopBar from './TopBar'

export default {
  title: 'Top Bar',
}

export const Usage = () => <TopBar />
