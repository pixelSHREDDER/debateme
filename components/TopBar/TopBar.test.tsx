import { describe, expect, test } from 'vitest'
import { render } from '@testing-library/react'
import TopBar from './TopBar'
import ProviderWrapper from '@/tests/helpers/ProviderWrapper'

describe('Top Bar component', () => {
  test('renders correctly', () => {
    const element = render(<TopBar />, { wrapper: ProviderWrapper })
    expect(element.container).toMatchSnapshot()
  });
})
