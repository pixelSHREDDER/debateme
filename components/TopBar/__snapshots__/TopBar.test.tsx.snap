// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Top Bar component > renders correctly 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-primary-color-filled: var(--mantine-color-orange-filled);--mantine-primary-color-filled-hover: var(--mantine-color-orange-filled-hover);--mantine-primary-color-light: var(--mantine-color-orange-light);--mantine-primary-color-light-hover: var(--mantine-color-orange-light-hover);--mantine-primary-color-light-color: var(--mantine-color-orange-light-color);--mantine-primary-color-0: var(--mantine-color-orange-0);--mantine-primary-color-1: var(--mantine-color-orange-1);--mantine-primary-color-2: var(--mantine-color-orange-2);--mantine-primary-color-3: var(--mantine-color-orange-3);--mantine-primary-color-4: var(--mantine-color-orange-4);--mantine-primary-color-5: var(--mantine-color-orange-5);--mantine-primary-color-6: var(--mantine-color-orange-6);--mantine-primary-color-7: var(--mantine-color-orange-7);--mantine-primary-color-8: var(--mantine-color-orange-8);--mantine-primary-color-9: var(--mantine-color-orange-9);}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-orange-4);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-orange-6);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <header
    class="_header_997776"
  >
    <div
      class="_inner_997776 m_7485cace mantine-Container-root"
      data-size="md"
      style="--container-size: var(--container-size-md);"
    >
      <div
        class="m_4081bf90 mantine-Group-root mantine-visible-from-xs"
        style="--group-gap: calc(0.3125rem * var(--mantine-scale)); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
      >
        <a
          class="mantine-focus-auto m_849cf0da _link_997776 m_b6d8b162 mantine-Text-root mantine-Anchor-root"
          data-active="true"
          data-underline="hover"
          href="/"
        >
          Home
        </a>
        <a
          class="mantine-focus-auto m_849cf0da _link_997776 m_b6d8b162 mantine-Text-root mantine-Anchor-root"
          data-underline="hover"
          href="/debate"
        >
          Debates
        </a>
      </div>
      <div
        class="m_4081bf90 mantine-Group-root mantine-visible-from-xs"
        style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: right; --group-wrap: wrap;"
      >
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: calc(0.0625rem * var(--mantine-scale)); --group-align: center; --group-justify: center; --group-wrap: wrap;"
        >
          <button
            aria-label="Toggle color scheme"
            class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
            data-size="lg"
            data-variant="default"
            style="--ai-size: var(--ai-size-lg); --ai-bg: var(--mantine-color-default); --ai-hover: var(--mantine-color-default-hover); --ai-color: var(--mantine-color-default-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-default-border);"
            type="button"
          >
            <span
              class="m_8d3afb97 mantine-ActionIcon-icon"
            >
              <svg
                class="tabler-icon tabler-icon-sun _icon_c853fd _light_c853fd"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 12m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0"
                />
                <path
                  d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7"
                />
              </svg>
              <svg
                class="tabler-icon tabler-icon-moon _icon_c853fd _dark_c853fd"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
                />
              </svg>
            </span>
          </button>
        </div>
        loading user....
      </div>
      <button
        class="mantine-focus-auto m_fea6bf1a mantine-Burger-root m_87cf2631 mantine-UnstyledButton-root mantine-hidden-from-xs"
        data-size="sm"
        style="--burger-size: var(--burger-size-sm);"
        type="button"
      >
        <div
          class="m_d4fb9cad mantine-Burger-burger"
          data-reduce-motion="true"
        />
      </button>
    </div>
  </header>
</div>
`;
